src/prusti/counter.ml

```diff
@@ -0,0 +1,26 @@
+
+(* https://github.com/FabianWolff/closure-examples/blob/master/counter.rs *)
+
+let foo f
+(*@ req f <: (fun v r (*@ ens r>=2 @*)) @*)
+= ()
+
+let counter_ok ()
+(* Here, we cannot prove that the obligation (foo inc) will pass *)
+= let x = ref 0 in
+  let inc ()
+  (*@ ex v; req x->v/\v>=0; ens x->v+1/\res=v @*)
+  = let r = !x in x := !x + 1; r in
+  inc ();
+  inc ();
+  (* assert (!x=2) *)
+  foo inc
+
+let counter_false () (* Entailment check is expected to fail **)
+(*@ ens T @*)
+= let x = ref 0 in
+  let inc ()
+  = let r = !x in x := !x + 1; r in
```

Comment:
I removed the specification for `inc` here because the [test report](https://github.com/EmilyOng/AlgebraicEffect/blob/21e63497cfe011f28979f37c7d62e6df277906b0/parsing/hiplib.ml#L754-L763) expects that functions with a `_false` suffix only sees false results.